cli/internal/fs/turbo_json.go

```diff
@@ -355,5 +363,14 @@ func (c *TurboJSON) UnmarshalJSON(data []byte) error {
 	c.Pipeline = raw.Pipeline
 	c.RemoteCacheOptions = raw.RemoteCacheOptions
 
+	if raw.TurboVersion != "" {
+		// possibly strict here
+		version, err := semver.StrictNewVersion(raw.TurboVersion)
```

Comment:
might not make a difference perf-wise but is this more than a regex match of `/\d+\.\d+\.\d+/`? (see canaries question in another comment also)